File: ch07/samp0703/main.js
const { OpenAIClient, AzureKeyCredential } = require('@azure/openai');
// Azure OpenAIのエンドポイントとAPIキーを設定
const endpoint = "https://sample-moonmile-openai.openai.azure.com/";
const apiKey = process.env["AZURE_OPENAI_API_KEY"] ;
const client = new OpenAIClient( endpoint, new AzureKeyCredential(apiKey) );

async function main() {
  try {
    const messages = [
      { role: "system", content: "You are an AI assistant that helps people find information." },
      { role: "user", content: "Azure OpenAIについて詳しく説明してください。" },
    ]
    const result = await client.getChatCompletions("test-x", messages, { 
        temperature: 0.7,
        maxTokens: 800,
        topP: 0.95,
        frequencyPenalty: 0.0,
        presencePenalty: 0.0,
    });
    console.log(result.choices[0].message.content);
  } catch (error) {
    console.error("Error:", error);
  }
}

// チャット関数を実行
main();

File: ch07/samp0703/main.py
#Note: The openai-python library support for Azure OpenAI is in preview.
#Note: This code sample requires OpenAI Python library version 0.28.1 or lower.

import os
import openai

openai.api_type = "azure"
openai.api_base = "https://sample-moonmile-openai.openai.azure.com/"
openai.api_version = "2023-07-01-preview"
openai.api_key = os.getenv("OPENAI_API_KEY")

message_text = [
  {"role":"system","content":"You are an AI assistant that helps people find information."},
  {"role":"user","content":"Azure OpenAIについて詳しく説明してください。"},
]

completion = openai.ChatCompletion.create(
  engine="test-x",
  messages = message_text,
  temperature=0.7,
  max_tokens=800,
  top_p=0.95,
  frequency_penalty=0,
  presence_penalty=0,
  stop=None
)


# 応答を表示します。
print(completion.choices[0].message['content'])
